crates/core/tedge_mapper/src/c8y/mapper.rs

```diff
@@ -153,13 +153,10 @@ impl TEdgeComponent for CumulocityMapper {
             cloud_config.set_clean_session(true);
 
             if use_certificate {
-                let cloud_broker_auth_config = tedge_config
-                    .mqtt_auth_config_cloud_broker(c8y_profile)
-                    .expect("error getting cloud broker auth config");
-                let client_config = cloud_broker_auth_config
-                    .to_rustls_client_config()
-                    .expect("error getting cloud broker auth config");
-                cloud_config.set_transport(Transport::tls_with_config(client_config.into()));
+                let tls_config = tedge_config
+                    .mqtt_client_config_rustls(c8y_config)
+                    .context("Failed to create MQTT TLS config")?;
+                cloud_config.set_transport(Transport::tls_with_config(tls_config.into()));
             } else {
                 // TODO(marcel): integrate credentials auth into MqttAuthConfig?
```

Comment:
Not related to this PR but we need to be careful to not introduce named TODO's into the code. Generally we like to avoid TODO's if possible all together, but sometimes they do make sense, but should not be tied to a specific person.

Reply:
Right, my intention in this case is to note some possible improvement to the code that for the time being I've avoided doing and noting down the author (me) to contact about this possible improvement, because the information about who's the author of the comment might change if the code is e.g. moved. So I'm not necessarily tying myself to that particular TODO item.

> Generally we like to avoid TODO's if possible all together
I'll keep it in mind and avoid adding any more TODOs.